File: user-service/src/main/java/rs/edu/raf/repository/UserRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.user.User;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User,Long> {
    Optional<User> findByEmailAndActiveIsTrue(String email);
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/repository/UserRepository.kt
package org.example.userservicekotlin.repository

import org.example.userservicekotlin.model.user.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByEmailAndActiveIsTrue(email: String): Optional<User>
}